modules/@apostrophecms/log/index.js

```diff
@@ -0,0 +1,255 @@
+          }
+        };
+      },
+      // Internal module, do not use it directly. See `@apostrophecms/module` for
+      // module level logging methods - logInfo, logError, etc.
+      logEntry(moduleSelf, severity, req, eventType, message, data) {
+        const args = self.processLoggerArgs(
+          moduleSelf,
+          severity,
+          req,
+          eventType,
+          message,
+          data
+        );
+        if (self.shouldKeepEntry(data)) {
+          self.apos.util[severity](...args);
+        }
+      },
+      // Do a minimal computation and validation of the arguemnts - logs should
+      // be fast. The function exepects 4 or 5 arguments (without counting the first optional `moduleSelf` argument),
+      // but all of the below will work:
+      // processLogArgs(moduleSelf, severity = 'info', eventType = 'event-type');
+      // processLogArgs(moduleSelf, severity = 'info', eventType = 'event-type', data = { module: 'my-module' });
+      // processLogArgs(moduleSelf, severity = 'info', eventType = 'event-type', message = 'message', data = { module: 'my-module' });
+      // processLogArgs(moduleSelf, severity = 'info', req, eventType = 'event-type', [...messageAndOrData]);
+      //
+      // If `moduleSelf` is provided, it is used to extract the module name from
+      // the `__meta` property.
+      // `severity` and `eventType` are required.
+      // `req`, `message` and `data` are optional.
+      // `req` is an apos request object. If provided, the `data` argument will be
+      // enriched with additional information from the request.
+      // `data.module` is recognized as a special property and is used to refortmat
+      // the message.
+      // `data.severity` is always set to the value of the `severity` argument.
+      // `data.type` is always set to the value of the `eventType` argument.
+      // message is optional, if not provided it is generated from the eventType and data.module.
+      // Optional message values:
+      // - `module: eventType: message`
+      // - `eventType: message`: (missing module)
+      // - `module: eventType`: (missing message)
+      // - `eventType`: (missing module and message)
+      //
+      // Returns [ data ] or [ message, data ] depending on option.messageAs.
+      // `data` is always an object containing at least a `type` and `severity` properties.
+      processLoggerArgs(...allArgs) {
+        const args = allArgs.slice(1);
+        const moduleSelf = allArgs[0];
+        let severity;
+        let req;
+        let eventType;
+        let message;
+        let data;
+
+        // Detect `req` argument with a simple duck type check - apos `req` object
+        // has always a translate `t` function.
+        if (args[1] && typeof args[1].t === 'function') {
+          [ severity, req, eventType, message, data ] = args;
+        } else {
+          [ severity, eventType, message, data ] = args;
+        }
+
+        if (!_.isString(eventType)) {
```

Comment:
Could be typeof, no?

Reply:
Sure, part of the cleanup (too much AI).